### PR TITLE
Enable components to get debug logging from our rest transports.

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

**Comment by mattmoor** [comment]
/hold

I want to try this downstream first :)

File: injection/sharedmain/main.go [hunk 1]
```diff
@@ -63,18 +64,39 @@ func GetConfig(masterURL, kubeconfig string) (*rest.Config, error) {
...
+	}
+
 	// If we have an explicit indication of where the kubernetes config lives, read that.
 	if kubeconfig != "" {
-		return clientcmd.BuildConfigFromFlags(masterURL, kubeconfig)
+		if c, err := clientcmd.BuildConfigFromFlags(masterURL, kubeconfig); err != nil {
+			return nil, err
+		} else {
+			return decorate(c), nil
+		}
```
Comment (vagababov) [inline injection/sharedmain/main.go:90]: unindent

Comment (mattmoor) [reply injection/sharedmain/main.go:91]: then I need to unbundle the declaration so that `c` is in scope....

**Pull request merged**